src/ir/properties.h

```diff
@@ -359,26 +358,28 @@ inline bool canEmitSelectWithArms(Expression* ifTrue, Expression* ifFalse) {
 // allocations, though, it doesn't matter what is in "..": there is nothing
 // in the wasm that we can check to find out if the results are the same or
 // not. (In fact, in this case they are always not the same.) So the
-// nondeterminism is "intrinsic."
+// nondeterminism is "intrinsic" and it is because each call to struct.new
+// generates a new value.
 //
-// Thus, loads are nondeterministic but not intrinsically so, while GC
-// allocations are actual examples of intrinsically nondeterministic
-// instructions. If wasm were to add "get current time" or "get a random number"
-// instructions then those would also be intrinsically nondeterministic.
+// Thus, loads are nondeterministic but not generative, while GC allocations
+// are in fact generative. Note that "generative" need not mean "allocation" as
+// if wasm were to add "get current time" or "get a random number" instructions
+// then those would also be generative - generating a new current time value or
+// a new random number on each execution, respectively.
 //
-//  * Note that NaN nondeterminism is ignored here. Technically that allows e.g.
-//    an f32.add to be nondeterministic, but it is a valid wasm implementation
-//    to have deterministic NaN behavior, and we optimize under that assumption.
-//    So NaN nondeterminism does not cause anything to be intrinsically
-//    nondeterministic.
+//  * Note that NaN nondeterminism is ignored here. While e.g. f32.add is
+//    nondeterministic due to NaNs, it does not generate a new value each time -
+//    on a single implementation it will return the same value all the time.
+//    And in any case, it is a valid wasm implementation to have deterministic
+//    NaN behavior, and we optimize under that assumption.
```

Comment:
```suggestion
//  * Note that NaN nondeterminism is ignored here. It is a valid wasm implementation to have deterministic
//    NaN behavior, and we optimize under that simplifying assumption.
```

The spec does not actually require that a single implementation always behave the same wrt NaNs.